src/Utils.ts

```diff
@@ -0,0 +1,106 @@
+ */
+
+import MetaProperty from "./MetaProperty";
+import { PathUtils } from "./PathUtils";
+import Constants from "./Constants";
+
+export class Utils {
+
+    /**
+     * Returns true if the editor is in Edit Mode
+     *
+     * @returns {Boolean} the result of the check of the Editor mode
+     */
+    public static isEditMode(): boolean {
+        const aemMode = PathUtils.getMetaPropertyValue(MetaProperty.WCM_MODE);
+        if (aemMode != null) {
+            return aemMode === Constants.AEM_MODE_EDIT;
+        }
+        else {
+            if (PathUtils.isBrowser()) {
+                const param = Utils.getEditParam();
+                return param === Constants.AEM_MODE_EDIT;
+            }
+            else {
+                return false;
+            }
+        }
+    }
+
+    /**
+     * Returns query parameter value for aem mode.
+     *
+     * @returns {String | null} the result of the check of the Editor mode
+     */
+    public static getEditParam() : string | null {
+        return new URL(document.location.href).searchParams.get(Constants.AEM_MODE_KEY);
+    }
+
+    /**
+     * Appends javascript to the page
+     */
+    public static appendScripts(clientLibUrl : string) : void {
+        const script = document.createElement('script');
+        const scripts = document.getElementsByTagName('script')[0];
+        script.src = clientLibUrl;
+        if (scripts.parentNode) {
+            scripts.parentNode.insertBefore(script, scripts);
+        }
+    }
+    /**
+     * Appends stylesheets to the page
+     */
+    public static appendStyleSheets(clientLibUrl : string) : void {
+        const link = document.createElement('link');
+        const links = document.getElementsByTagName('link')[0];
+        link.href = clientLibUrl;
+        if (links.parentNode) {
+            links.parentNode.insertBefore(link, links);
+        }
+    }
+
+    /**
+     * Generates clientlib url
+     */
+    public static generateClientLibsUrl(domain : string) : string[] {
+        const clientlibs: string[] =  Array<string>();
+        const path = '/etc.clientlibs/cq/gui/components/authoring/editors/clientlibs/internal/';
+        clientlibs.push(`${domain}${path}page.js`);
+        clientlibs.push(`${domain}${path}page.css`);
+        clientlibs.push(`${domain}${path}pagemodel/messaging.js`);
+        clientlibs.push(`${domain}${path}messaging.js`);
+        return clientlibs;
+    }
+
+    /**
+     * Returns true if editor loaded remotely
+     */
+    public static appendClientLibs(domain : string) : void {
+        const clientLibUrls = Utils.generateClientLibsUrl(domain);
+        clientLibUrls.forEach(lib => {
+            if (lib.endsWith('.js')) {
+                this.appendScripts(lib);
+            }
+            else {
+                this.appendStyleSheets(lib);
+            }
+        });
+    }
+
+    /**
+     * Returns true if editor loaded remotely
+     */
+    public static isAsync(domain : string) : boolean {
```

Comment:
The name and the behavior of the function don't seem to match. What about something like `(is|does)DomainEquals`, `isCurrentDomain`, `isSameDomain`, etc..? And if necessary reverse the test

Reply:
Are you certain that what you are looking for is the `pathname` and not the `window.location.host` or other property that is closer to the actual domain?

https://url.spec.whatwg.org/#concept-domain

Reply:
If you are using `window`, once again, be sure to consider the context of SSR and isomorphism

Reply:
The check of the same or different domain cannot be done server-side unless we have the information if `host` domain. We may use something similar to the way we selected `aemmode`[0] where we add one more property to the config and then default to it. @grubyak wdyt. 
[0] https://github.com/adobe/aem-spa-page-model-manager/blob/00f6a05e0455abbb2f6eade7de15a2cb689bb48a/src/ModelManager.ts#L59